### DISCUSSION TITLE
[SOLVED] Is it possible to use wimboot without ipxe, only with syslinux, dnsmasq and tftpd-hpa?

### DISCUSSION THREAD

**Discussion by ghost** [ipxe/ipxe#675] [Q&A] [answered]
Hello,

I am trying to set up a PXE server with above packages but always get the following error from syslinux: 
```
Cannot relocate kernel, bailing out
Booting kernel failed: Bad file number
```

My syslinux.cfg is:

```
LABEL WINPE
MENU LABEL Windows PE
KERNEL wimboot
INITRD winpe/Boot/BCD,winpe/Boot/boot.sdi,winpe/sources/boot.wim
```
`root@debianpxe:/var/tftp# file wimboot
wimboot: Linux kernel x86 boot executable bzImage, version v2.7.4, RW-rootFS,`

What can I do to boot Windows PE? 

Unfortunately, iPXE is not an option.

Thanks

**Comment by NiKiZe** [comment, marked as answer]
Why is iPXE not an option? (It might help us understand the limitations you are facing better)
Which version of syslinux are you using? Are you not using pxelinux?
dnsmasq and tftpd-hpa should be irrelevant.

For pcbios, I think you should be able to create a cpio archive that has the files needed, which then wimboot uses.
But looking at your initrd, you can drop both BCD and boot.sdi, that only leaves boot.wim.

So back to the error, do you get the same error if you only try to run wimboot? having wimboot execute properly is a good first step before throwing the hugeish (especially considering over tftp) .wim on it.
Also try to run wimboot with the debug options.

**Reply by ghost** [reply]
My platform ist x86_64 UEFI and I'm using syslinux.efi.

`root@debianpxe:/var/tftp# file syslinux.efi
syslinux.efi: PE32+ executable (EFI application) x86-64 (stripped to external PDB), for MS Windows
root@debianpxe:/var/tftp# apt list syslinux
Auflistung… Fertig
syslinux/stable,now 3:6.04~git20190206.bf6db5b4+dfsg1-3+b1 amd64  [installiert]`

I cannot use iPXE because I understand that a proper configuration is only possible with isc dhcpd. And I seem to remember that my DHCP config could not be implemented with it, so I had to switch to dnsmasq. That means I use a dedicated DHCP server and use the dhcp-proxy function of dnsmasq to provide PXE.

I also tried to start wimboot-only but it is running in the same error.

**Reply by NiKiZe** [reply]
Efi version of syslinux does not create the structure needed for wimboot.
Also it seems like it tries to execute as a kernel, instead of as an efi binary.

I, and many others, have working dnsmasq configurations. Especially proxy-dhcp https://ipxe.org/appnote/proxydhcp

I would suggest that you read existing discussions on dnsmasq, and feel free to ask more questions if needed. (New topics is probably best)

**Reply by ghost** [reply]
I can state this problem as solved, I found this patched wimboot: https://github.com/Sporesirius/syslinuxefi-wimboot/blob/master/EFI/BOOT/wimboot

And its doing exactly what I want.

**Comment by stappersg** [comment]
On Mon, Jul 11, 2022 at 01:18:22PM -0700, anon0815 wrote:
> I cannot use iPXE because I understand that a proper configuration is
> only possible with isc dhcpd. And I seem to remember that my DHCP config
> could not be implemented with it, so I had to switch to dnsmasq. That
> means I use a dedicated DHCP server and use the dhcp-proxy function
> of dnsmasq to provide PXE.

That would be a much better discussion topic for the iPXE community.

**Reply by ghost** [reply]
I'm sorry, I chose the wrong repository. My fault.
